### PR TITLE
Add plot_canonical

### PR DESCRIPTION
**Pull request by JedGrabman**
Adding plot_canonical from `model_tools` vignette (renamed from `canonical_plot` to match evalcast naming conventions).

### CODE REVIEW

File: R-packages/evalcast/R/plot_canonical.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#' @param dots when set to TRUE, a dot layer is included in the plot
+#' @param lines when set to TRUE, a line layer is included in the plot
+#' @param group_vars The set of variable names in `df` which to aggregate over.
+#'   "forecaster" by default.
+#' @param facet_rows A variable name to facet data over. Creates a
+#'   separate row of plots for each value of specified variable. Can be used
+#'   with `facet_cols` to create a grid of plots. 
+#' @param facet_cols Same as `facet_rows`, but with columns.
+#' @param base_forecaster If set, scales they y-value for all forecasters by
+#'   the corresponding values of the base_forecaster
```
Comment (sgsmob) [inline R-packages/evalcast/R/plot_canonical.R:16]: wrap in backticks

File: R-packages/evalcast/R/plot_canonical.R [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#' @param scale_before_aggregation If TRUE, scales results by the
+#'   `base_forecaster` before aggregating by the `aggr` function
+#' @param title title of the plot, if set
+#' @param subtitle subtitle of the plot, if set
+#' @param xlab x-axis label, if set
+#' @param ylab y-axis label, if set
+#' @param legend_position where to position the legend ("bottom" by default)
+#' @param legend_title title of the legend, if set
+#'
```
Comment (sgsmob) [inline R-packages/evalcast/R/plot_canonical.R:25]: Can drop these blank lines

**Review by sgsmob — APPROVED** [review]
Just 2 minor formatting issues.

File: R-packages/evalcast/R/plot_canonical.R [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+#' @param lines when set to TRUE, a line layer is included in the plot
+#' @param group_vars The set of variable names in `df` which to aggregate over.
+#'   "forecaster" by default.
+#' @param facet_rows A variable name to facet data over. Creates a
+#'   separate row of plots for each value of specified variable. Can be used
+#'   with `facet_cols` to create a grid of plots. 
+#' @param facet_cols Same as `facet_rows`, but with columns.
+#' @param base_forecaster If set, scales they y-value for all forecasters by
+#'   the corresponding values of the base_forecaster
+#' @param scale_before_aggregation If TRUE, scales results by the
```
Comment (dajmcdon) [inline R-packages/evalcast/R/plot_canonical.R:17]: Let's mention that this is FALSE by default.

**Review by dajmcdon — CHANGES_REQUESTED** [review]
Just one minor addition to @sgsmob 's requests.

**Comment by dajmcdon** [comment]
@JedGrabman once we merge this, can you fix the [modeltools vignette](https://github.com/cmu-delphi/covidcast/tree/main/R-packages/modeltools/vignettes) to use this function and the built in WIS, AE funs?

File: R-packages/evalcast/R/plot_canonical.R (hunk 1 above)
Comment (JedGrabman) [reply R-packages/evalcast/R/plot_canonical.R]: Done

File: R-packages/evalcast/R/plot_canonical.R (hunk 2 above)
Comment (JedGrabman) [reply R-packages/evalcast/R/plot_canonical.R]: Done

File: R-packages/evalcast/R/plot_canonical.R (hunk 3 above)
Comment (JedGrabman) [reply R-packages/evalcast/R/plot_canonical.R:17]: Done

**Review by dajmcdon — APPROVED** [review]

**Pull request merged**